Add archivedAt column and update repositories

hackertronix opened the pull request:
A couple of things that need clarification

- I have updated the raw SQL queries for `nonDeletedAndNonEmptyNotes ` and `nonDeletedNotes ` but I think the query name should also reflect that let me know if I should change and those as well

- the `markAsArchived` implementation in FakeNoteRepository will need a Clock instance which can be added as a constructor param but I think a better approach would be to add a dedicated function in the FakeData class to handle archiving for tests, what do you think?

hackertronix committed:
Add archivedAt column and update repositories

saket commented on shared/src/commonMain/sqldelight/me/saket/press/data/shared/Note.sq:
```diff
@@ -7,6 +7,7 @@ CREATE TABLE note (
   content TEXT NOT NULL,
   createdAt TEXT AS DateTime NOT NULL,
   updatedAt TEXT AS DateTime NOT NULL,
+  archivedAt TEXT AS DateTime,
```
This unfortunately cannot be a nullable DateTime column because of this issue: https://youtrack.jetbrains.com/issue/KT-35234

hackertronix replied:
Shall I make this a  string in that case just like the `deletedAtString ` column?

saket replied:
Yes please!

saket commented on shared/src/commonTest/kotlin/me/saket/press/shared/note/FakeNoteRepository.kt:
```diff
@@ -56,4 +56,9 @@ class FakeNoteRepository : NoteRepository {
+
+  override fun markAsArchived(noteUuid: Uuid): Completable {
+   TODO()
+    //We need the instance of a Clock to set the archivedAt property
```
This isn't needed. A "fake" repository is used for testing other classes that depend on it, where you don't care what the repository does as long as the class under test is calling the right APIs on the repository. If it were an integration test then you would pass the real repository. 

Does this make sense? I can explain further if you want. You can simply copy the `markAsDeleted()` function here.

hackertronix replied:
Thanks for clarifying it makes sense :)

saket commented on shared/src/commonMain/kotlin/me/saket/press/shared/note/RealNoteRepository.kt:
```diff
@@ -81,4 +82,13 @@ internal class RealNoteRepository(
+          archivedAt = clock.nowUtc()
+      )
+    }
+  }
```
This is missing a test. I just pushed a test to master for checking `markAsDeleted()` that you can use as a reference.

saket commented on shared/src/commonMain/sqldelight/me/saket/press/data/shared/Note.sq:
```diff
@@ -7,6 +7,7 @@ CREATE TABLE note (
   content TEXT NOT NULL,
   createdAt TEXT AS DateTime NOT NULL,
   updatedAt TEXT AS DateTime NOT NULL,
+  archivedAt TEXT AS DateTime,
```
This is missing a SQL migration. The app will 💥 for users who install this update. https://github.com/cashapp/sqldelight#migrations

saket commented:
> I have updated the raw SQL queries for nonDeletedAndNonEmptyNotes and nonDeletedNotes but I think the query name should also reflect that let me know if I should change and those as well

Agree with you. I can't think of a better name right now so let's just rename them to `notes` and `nonEmptyNotes`.

> the markAsArchived implementation in FakeNoteRepository will need a Clock instance which can be added as a constructor param but I think a better approach would be to add a dedicated function in the FakeData class to handle archiving for tests, what do you think?

Answered here: https://github.com/saket/press/pull/21#discussion_r356938407

hackertronix committed:
Made requested changes to PR

1. Add a migration file
2. Change type of archivedAt column
3. Add tests for both repositories

hackertronix commented:
Hey, @saket I've made the required changes.

hackertronix committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	shared/src/commonTest/kotlin/me/saket/press/shared/note/RealNoteRepositoryTest.kt

hackertronix committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	shared/src/commonTest/kotlin/me/saket/press/shared/note/RealNoteRepositoryTest.kt

hackertronix committed:
Merge branch 'master' of github.com:hackertronix/Press

hackertronix committed:
Update checks.yml

saket commented:
Making some changes to this and merging your commit to master manually: https://github.com/saket/press/commit/9af28191c66e0a76fb84a8a1bba6e3a17c7a6ff4
